.github/workflows/deploy.yml

```diff
@@ -144,6 +144,16 @@ jobs:
         run: hugo -v --ignoreCache
         working-directory: web
 
+      - name: Copy nightly landing page as index.html
+        if: github.repository_owner == 'theforeman' && github.ref == 'refs/heads/master'
+        run: cp -f nightly.html nightly/index.html
+        working-directory: web/public
+
+      - name: Copy stable landing page as index.html
+        if: github.repository_owner == 'theforeman' && (startsWith(github.ref, 'refs/heads/2.') || startsWith(github.ref, 'refs/heads/3.') || startsWith(github.ref, 'refs/heads/4.'))
+        run: cp -f ${{ env.BRANCH_NAME }}.html ${{ env.BRANCH_NAME }}/index.html
```

Comment:
I checked out the 2.5 branch and ran `hugo` but there I don't see a `2.5.html` file. How would that be built?

Reply:
The whole web/ directory is a static site, it should pick the `web/content/2.5.md` file and generate the HTML.

Note it is always built from master. Just use master. I think our GHA is confusing - we build from the branch which is being built and copy files over not deleting contents of the result directory, that's why we still see 2.5 even after GHA with an older contents is triggered. I think we should change the action to always pull `master` to build the web portion of the site.

Reply:
Yeah let's fix this: https://github.com/theforeman/foreman-documentation/pull/590

Reply:
That is now merged. Does that mean changes to this PR are now needed? Since we now always build from master, does it mean we can somehow teach hugo to do this for us instead of manually working around it?

Reply:
This is still needed, the problem here is that hugo is used to build the whole site (main index, css, pages etc) while in the root directory we also have `/X.Y` directories which are generated by a separate workflow (`asciidoctor`). And these directories are not managed by hugo, I would like not to go there as I found hugo quite complex, way more complex than jekyll. I was hoping it was more simple, it advertises itself as a lightweight but its not.

One day I was even thinking ditching it completely for just Ruby ERB, we have a such a simple use case that it is probably an overkill.